### PR TITLE
migrator: schema validation of records

### PR DESCRIPTION
**Pull request by jalavik**
* Adds `invenio-jsonschemas` as dependency to validate records
  upon migration. Errors are non-breaking and only logged.

* Adds upgrade script for `inspire_prod_records` table.

* Moves schemas to new base/jsonschemas folder to be discovered
  by `invenio-jsonschemas`.

* Adds `$schema` key to all generated dojson output pointing to
  the appropriate schema for each document type (collection).

It reveals several validation errors with our demo records:

```
2016-02-09 15:08:52,242 WARNING: ValidationError: Record 1319638: [{'source': 'arXiv', 'title': 'Precision luminosity measurements at LHCb'}, {'source': 'arXiv', 'title': 'Precision luminosity measurements at LHCb'}] has non-unique elements

Failed validating u'uniqueItems' in schema[u'properties'][u'titles']:
    {u'items': {u'properties': {u'source': {u'type': u'string'},
                                u'subtitle': {u'type': u'string'},
                                u'title': {u'type': u'string'}},
                u'type': u'object'},
     u'type': u'array',
     u'uniqueItems': True}

On instance[u'titles']:
    [{'source': 'arXiv',
      'title': 'Precision luminosity measurements at LHCb'},
     {'source': 'arXiv',
      'title': 'Precision luminosity measurements at LHCb'}]
```

```
2016-02-09 15:08:46,662 WARNING: ValidationError: Record 1317658: '1' is not of type u'integer'

Failed validating u'type' in schema[u'properties'][u'references'][u'items'][u'properties'][u'number']:
    {u'type': u'integer'}

On instance[u'references'][0][u'number']:
    '1' [in /Users/jlavik/Envs/next/src/inspire/inspirehep/modules/migrator/tasks.py:356]
```

```
2016-02-09 15:08:39,800 WARNING: ValidationError: Record 1313115: ['Invenio/1.1.2.1260-aa76f refextract/1.5.44', 'content.pdf;1'] is not of type u'string'

Failed validating u'type' in schema[u'properties'][u'refextract'][u'items'][u'properties'][u'version']:
    {u'type': u'string'}

On instance[u'refextract'][0][u'version']:
    ['Invenio/1.1.2.1260-aa76f refextract/1.5.44', 'content.pdf;1'] [in /Users/jlavik/Envs/next/src/inspire/inspirehep/modules/migrator/tasks.py:356]
```

### CODE REVIEW

File: inspirehep/dojson/utils.py [hunk 1]
```diff
@@ -21,6 +21,20 @@
...
+
+    def __init__(self, schema=None, *args, **kwargs):
+        super(SchemaOverdo, self).__init__(*args, **kwargs)
+        self.schema = schema
+
+    def do(self, blob):
+        output = super(SchemaOverdo, self).do(blob)
+        output['$schema'] = self.schema
+        return output
+
```
Comment (kaplun) [inline inspirehep/dojson/utils.py]: This is duplicate from `schema.py`.

Comment (jalavik) [inline inspirehep/dojson/utils.py]: Ah! thanks. yeah, I was moving back and forth like a :dancer:

**Comment by jmartinm** [comment]
:+1:  Looks good to me.

**Comment by jalavik** [comment]
Kewl. We can collect the types of validation errors we see to a new issue.

**Comment by kaplun** [comment]
One always learn something new everyday:
http://www.urbandictionary.com/define.php?term=kewl

**Pull request merged**